docs/resources/Channel.md

```diff
@@ -261,38 +261,39 @@ Represents a message sent in a channel within Discord.
+| thread?                     | [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object                                                                           | the thread that was started from this message, includes [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) object                                                                                                                                            |
+| components?\*\*             | array of [message components](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/component-object)                                              | sent if the message contains components like buttons, action rows, or other interactive components                                                                                                                                                                      |
+| sticker_items?              | array of [message sticker item objects](#DOCS_RESOURCES_STICKER/sticker-item-object)                                               | sent if the message contains stickers                                                                                                                                                                                                                                   |
+| stickers?                   | array of [sticker](#DOCS_RESOURCES_STICKER/sticker-object) objects                                                                 | **Deprecated** the stickers sent with the message                                                                                                                                                                                                                       |
 | position?                   | integer                                                                                                                            | A generally increasing integer (there may be gaps or duplicates) that represents the approximate position of the message in a thread, it can be used to estimate the relative position of the message in a thread in company with `total_message_sent` on parent thread |
+| role_subscription_data?     | [role subscription data](#DOCS_RESOURCES_CHANNEL/role-subscription-data-object) object                                             | data of the role subscription purchase or renewal that prompted this ROLE_SUBSCRIPTION_PURCHASE message                                                                                                                                                                 |
```

Comment:
this is correct _for now_, we actually wanted to give it a generic name so that we can store any role subscription data on a message in one place. Right now we're using it to show the purchase recognition message but in the future, we might use this field to store some other role subscription message data. 

We can update this in the future if we use it for something else though 👍

Reply:
If this were changed how would an application deal with it without crashing?

Reply:
The precedent for volatile internal fields is to prefix the name with `_`, similar to `_trace` in the READY dispatch.

Reply:
should I mark any of the fields as optional and/or nullable?

Reply:
> If this were changed how would an application deal with it without crashing?

what I mean is that we might add new fields to that `role_subscription_data` for a new message type in the future and at that point, this field wouldn't be only used for the `ROLE_SUBSCRIPTION_PURCHASE` message type. we definitely won't change anything in the `role_subscriptions_data` for the `ROLE_SUBSCRIPTION_PURCHASE` type.

Reply:
if a field is not marked as optional then libraries would assume that the field will always be present in role_subscription_data, and might error if that is not true

Reply:
actually, seems like my memory is failing me, looks like we didn't mark any of those fields optional on the clientside so ignore my comments